skiplang/prelude/ts/src/sk_mem_utils.ts

```diff
@@ -108,14 +109,18 @@ export class MemFS implements FileSystem {
   }
 
   writeToFile(fd: int, content: string) {
-    this.files[fd].write(content);
+    let file = this.files[fd];
+    if (file == undefined) {
```

Comment:
Probably `===` instead of `==` is better here but it shouldn't matter much.

Reply:
This made me think a bit more about the various notions of equality, and I changed this to use `??` instead of either `== undefined` or `=== undefined`. My understanding is that `e ?? fail(msg)` evaluates to `e` if it is neither `null` nor `undefined` and otherwise evaluates `fail(msg)`. Do you think that is an appropriate semantics here? If not, it is probably not appropriate in all the other places where `?? fail` is used in this PR.

Reply:
Yes. I was concerned about arrays where the element type admits `null` and checked a bunch of them, but I should probably make a pass over them all.